libswc/seat.c

```diff
@@ -83,58 +43,39 @@ handle_keyboard_focus_event(struct wl_listener *listener, void *data)
 		struct wl_client *client = wl_resource_get_client(event_data->new->surface->resource);
 
 		/* Offer the selection to the new focus. */
-		data_device_offer_selection(seat->base.data_device, client);
+		data_device_offer_selection(seat->data_device, client);
 	}
 }
 
-static void
-handle_data_device_event(struct wl_listener *listener, void *data)
+void
+seat_handle_data_device_event(struct wl_listener *listener, void *data)
 {
-	struct seat *seat = wl_container_of(listener, seat, data_device_listener);
+	struct swc_seat *seat = wl_container_of(listener, seat, data_device_listener);
 	struct event *ev = data;
 
 	if (ev->type != DATA_DEVICE_EVENT_SELECTION_CHANGED)
 		return;
 
-	if (seat->base.keyboard->focus.client)
-		data_device_offer_selection(seat->base.data_device, seat->base.keyboard->focus.client);
-}
-
-static void
-handle_swc_event(struct wl_listener *listener, void *data)
-{
-	struct seat *seat = wl_container_of(listener, seat, swc_listener);
-	struct event *ev = data;
-
-	switch (ev->type) {
-	case SWC_EVENT_DEACTIVATED:
-		libinput_suspend(seat->libinput);
-		keyboard_reset(seat->base.keyboard);
-		break;
-	case SWC_EVENT_ACTIVATED:
-		if (libinput_resume(seat->libinput) != 0)
-			WARNING("Failed to resume libinput context\n");
-		break;
-	}
+	if (seat->keyboard->focus.client)
+		data_device_offer_selection(seat->data_device, seat->keyboard->focus.client);
 }
 
 /* Wayland Seat Interface */
+
 static void
 get_pointer(struct wl_client *client, struct wl_resource *resource, uint32_t id)
 {
-	struct seat *seat = wl_resource_get_user_data(resource);
+	struct swc_seat *seat = wl_resource_get_user_data(resource);
 
-	if (!pointer_bind(&seat->pointer, client, wl_resource_get_version(resource), id))
-		wl_resource_post_no_memory(resource);
```

Comment:
Isn't this error check still needed?

Reply:
You're right, I'm not sure how that disappeared...